lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -689,6 +689,28 @@ void mlir::triton::asyncLaunchDots(scf::ForOp forOp) {
               !CArg.hasOneUse())
             valid = false;
 
+          // Both operands needs to have been multi-buffered if they come from
+          // shared memory otherwise we will have a race condition as the shared
+          // memory will be overriden before the wgmma wait.
+          // Note that we relay on the assumption that `createAsynOps` will have
+          // allocated enough buffers to pipeline dot operation with 2 stages.
```

Comment:
What does it mean "dot operation with 2 stages."?

Reply:
I get it now. 
You meant we just issued an async dot, but `async_wait` is waiting for another async dot.
Maybe just `dot operation`->`async dot operation`